src/main/java/dev/openfeature/sdk/providers/memory/InMemoryProvider.java

```diff
@@ -0,0 +1,166 @@
+package dev.openfeature.sdk.providers.memory;
+
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.ProviderEventDetails;
+import dev.openfeature.sdk.ErrorCode;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+/**
+ * In-memory provider.
+ */
+@Slf4j
+public class InMemoryProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "InMemoryProvider";
+
+    private Map<String, Flag<?>> flags;
+
+    @Getter
+    private ProviderState state = ProviderState.NOT_READY;
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    public InMemoryProvider(Map<String, Flag<?>> flags) {
+        this.flags = new HashMap<>(flags);
```

Comment:
Good idea to copy this ref.

Reply:
Sorry, I wasn't clear. I was saying the way you did it was a good idea instead of copying it. It's a good idea exactly because it prevents the bug :). You already did it right :smile: